cv.typ

```diff
@@ -2,341 +2,391 @@
-                    *#link(pub.url)[#pub.name]* \
-                ] else [
-                    *#pub.name* \
-                ]
-                // line 2: publisher and date
-                Published on #text(style: "italic")[#pub.publisher]  #h(1fr) #date \
-            ]
-        }
-    ]}
+  if info.at("publications", default: none) != none {
+    block({
+      [== Research & Publications]
+
+      for pub in info.publications {
+        let url = pub.at("url", default: none)
+        let date = utils.str_to_date(pub.releaseDate)
+
+        block(width: 100%, breakable: isbreakable, {
+          // line 1: publication title
+          line_one(pub.name, url: url)
+
+          linebreak()
+          // line 2: publisher and date
+          [Published on #line_two(pub.publisher, right_end: date)]
+        })
+      }
+    })
+  }
 }
 
 #let cvskills(info, isbreakable: true) = {
-    if (info.languages != none) or (info.skills != none) or (info.interests != none) {block(breakable: isbreakable)[
-        == Skills, Languages, Interests
-        #if (info.languages != none) [
-            #let langs = ()
-            #for lang in info.languages {
-                langs.push([#lang.language (#lang.fluency)])
-            }
-            - *Languages*: #langs.join(", ")
-        ]
-        #if (info.skills != none) [
-            #for group in info.skills [
-                - *#group.category*: #group.skills.join(", ")
-            ]
-        ]
-        #if (info.interests != none) [
-            - *Interests*: #info.interests.join(", ")
-        ]
-    ]}
+  if (info.at("languages", default: none) != none) or (info.at("skills", default: none) != none) or (info.at("interests", default: none) != none) {
+    block(breakable: isbreakable, {
+      [== Skills, Languages, Interests]
+
+      let langs = info.at("languages", default: none)
+      let langs_join = ()
+      for lang in langs {
+        langs_join.push([#lang.language (#lang.fluency)])
+      }
+
+      dot_list("Languages", langs_join)
+
+      let skills = info.at("skills", default: none)
+      for skill in skills {
+        dot_list(skill.category, skill.skills)
+      }
+
+      let interests = info.at("interests", default: none)
+      dot_list("Interests", interests)
+    })
+  }
 }
 
 #let cvreferences(info, isbreakable: true) = {
-    if info.references != none {block[
-        == References
-        #for ref in info.references {
-            block(width: 100%, breakable: isbreakable)[
-                #if ref.url != none [
-                    - *#link(ref.url)[#ref.name]*: "#ref.reference"
-                ] else [
-                    - *#ref.name*: "#ref.reference"
-                ]
-            ]
-        }
-    ]} else {}
+  if info.at("references", default: none) != none {
+    block({
+      [== References]
+
+      for ref in info.references {
+        let foo = ()
+        foo.push(ref.reference)
+        dot_list(ref.name, foo, url: ref.url)
```

Comment:
May I ask if `foo` here is needed? Could it instead be an [array created directly](https://typst.app/docs/reference/foundations/array/#constructor)?

```suggestion
        dot_list(ref.name, (ref.reference,), url: ref.url)
```